babi.py

```diff
@@ -914,7 +914,82 @@ def _get_char(stdscr: 'curses._CursesWindow') -> Key:
     return Key(wch, key, keyname)
 
 
-EditResult = enum.Enum('EditResult', 'EXIT NEXT PREV')
+EditResult = enum.Enum('EditResult', 'EXIT NEXT PREV EDIT')
+
+
+class Command:
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        keybinds: List[str],
+        func: Callable[[Screen, Key], EditResult],
+        aliases: List[str] = [],
```

Comment:
this is a mutable default which has a problem. consider the following:

```python
def f(x=[]):
    x.append(1)
    print(x)
```

```pycon
>>> f()
[1]
>>> f()
[1, 1]
>>> f()
[1, 1, 1]
```

Reply:
Why is that a problem? It allows programmatic setting of aliases for commands that don't already have them

Reply:
Using `[]` as the default for `aliases` causes different instances of the object to share the same object for `aliases`. In the above example, the `x` parameter ends up being shared across all calls to `f()`. https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Reply:
Could I set it to None then until it's assigned inside the decorator? @asottile @mxr

Reply:
Yes or use tuples and set the default to `()`

Reply:
The problem with that is that it breaks programmatic setting of keybinds.

Reply:
It depends how you want to support programmatic keybinds. If you want to support keybinds with restarting the editor it's ok: without restarting the editor, then yes it doesn't work, but that can be a v2 feature added later